### ISSUE TITLE
AutoFactory: Incompatibility truth 1.0.1 and compile-testing 0.18

### ISSUE THREAD

**Issue by frankbenoit** [google/auto#891]
The library truth changed the API of `Subject`:
.. 0.45: Subject has generic args (https://truth.dev/api/0.45/com/google/common/truth/Subject.html)
0.46 .. 1.0.1: Subject has no generic args (https://truth.dev/api/0.46/com/google/common/truth/Subject.html)

The compile-testing library:
today last release 0.18: makes use of Subject with generic args. The current source has the generic args already removed, but he sources of 0.18 release still have them. Also, this 0.18 defines the dependency to truth 0.44

In auto-factory, the version for truth is set to 1.0.1. This causes to show compilation errors, at least with Eclipse and VS Code.
To get the project setup in Eclipse and being able to run unit tests, I did these changes:

- set truth to 0.45
- add dependency to javax.annotation-api (to compile the generated `@Generated` annotation)
- (in Eclipse ensure the installed JRE adds the tools.jar)

```
    <truth.version>0.45</truth.version>
    ...
    <dependency>
      <groupId>javax.annotation</groupId>
      <artifactId>javax.annotation-api</artifactId>
      <version>1.3.2</version>
    </dependency>
```

**Comment by netdpb** [comment]
I think it makes more sense to release a new version of Compile Testing and then depend on that.

**Comment by frankbenoit** [comment]
I created this one: https://github.com/google/compile-testing/issues/195

**Comment by netdpb** [comment]
Thanks.

**Comment by frankbenoit** [comment]
compile-testing 0.19 release is available now.

**Comment by frankbenoit** [comment]
fixed with 14bb4c20c792b0ea16dde838ae10e9329f05cda4
